app/models/request_item.rb

```diff
@@ -1,17 +1,22 @@
 class RequestItem
-  attr_accessor :name, :quantity, :on_hand
+  attr_accessor :name, :quantity, :on_hand, :on_hand_for_location
 
-  def self.from_json(json, organization)
+  def self.from_json(json, organization, location)
     item = Item.find(json['item_id'])
     quantity = json['quantity']
     on_hand = organization.inventory_items.where(item_id: item.id).sum(:quantity)
-
-    new(item.name, quantity, on_hand)
+    on_hand_for_location = if location
+      location.inventory_items.where(item_id: item.id).sum(:quantity)
```

Comment:
I'am not 100% sure but isn't there only 1 record of an item per location? In that case I'am not sure if `.where` and `.sum` make sense if it is just one record?

Reply:
I *think* you're right, but there doesn't seem to be an outright validation enforcing this. @seanmarcia can you comment?

Reply:
Good question. I'd have to dig into it. It definitely feels like that is correct.